infra/proposer.go

```diff
@@ -17,13 +18,28 @@ type Proposers struct {
 	logger *log.Logger
 }
 
-func CreateProposers(conn, client int, nodes []Node, crypto *Crypto, logger *log.Logger) *Proposers {
+func GetTLSCACerts(files string) ([][]byte, error) {
+	var certs [][]byte
+	in, err := ioutil.ReadFile(files)
+	if err != nil {
+		return nil, err
+	}
+
+	certs = append(certs, in)
+	return certs, nil
+}
+
+func CreateProposers(conn, client int, nodes []Node, logger *log.Logger) *Proposers {
 	var ps [][]*Proposer
 	//one proposer per connection per peer
 	for _, node := range nodes {
 		row := make([]*Proposer, conn)
 		for j := 0; j < conn; j++ {
-			row[j] = CreateProposer(node.Addr, crypto, logger)
+			tls, err := GetTLSCACerts(node.TLSCACert)
```

Comment:
we don't need to repeatedly read in the **same** file here, right?

Reply:
yep, should be node level/